modules/apps/object/object-test/src/testIntegration/java/com/liferay/object/service/test/ObjectFolderItemLocalServiceTest.java

```diff
@@ -0,0 +1,355 @@
+				objectFolder.getObjectFolderId(),
+				_objectDefinitionLocalService);
+
+		_assertObjectFolderItems(
+			new Long[] {objectDefinition.getObjectDefinitionId()},
+			objectFolder.getObjectFolderId());
+
+		ObjectRelationshipTestUtil.addObjectRelationship(
+			_objectRelationshipLocalService, _objectDefinition4,
+			objectDefinition);
+
+		_assertObjectFolderItems(
+			new Long[] {
+				objectDefinition.getObjectDefinitionId(),
+				_objectDefinition1.getObjectDefinitionId(),
+				_objectDefinition4.getObjectDefinitionId()
+			},
+			_objectFolderC.getObjectFolderId());
+
+		_assertObjectFolderItems(
+			new Long[] {
+				objectDefinition.getObjectDefinitionId(),
+				_objectDefinition4.getObjectDefinitionId()
+			},
+			objectFolder.getObjectFolderId());
+
+		_objectDefinitionLocalService.deleteObjectDefinition(objectDefinition);
+
+		_objectFolderLocalService.deleteObjectFolder(objectFolder);
+	}
+
+	@Test
+	public void testDeleteObjectFolderItem() throws Exception {
+
+		// ObjectFolderItem is not deleted when object definition is unlinked
+
+		_objectFolderItemLocalService.deleteObjectFolderItem(
+			_objectDefinition2.getObjectDefinitionId(),
+			_objectFolderB.getObjectFolderId());
+
+		Assert.assertNotNull(
+			_objectFolderItemLocalService.getObjectFolderItem(
+				_objectDefinition2.getObjectDefinitionId(),
+				_objectFolderB.getObjectFolderId()));
+
+		// ObjectFolderItem is not deleted when related object definition is
+		// unlinked
+
+		_objectFolderItemLocalService.deleteObjectFolderItem(
+			_objectDefinition1.getObjectDefinitionId(),
+			_objectFolderB.getObjectFolderId());
+
+		Assert.assertNotNull(
+			_objectFolderItemLocalService.getObjectFolderItem(
+				_objectDefinition1.getObjectDefinitionId(),
+				_objectFolderB.getObjectFolderId()));
+
+		// ObjectFolderItem is deleted when is linked and related object
+		// definition is linked
+
+		ObjectFolderItem objectFolderItem =
+			_objectFolderItemLocalService.addObjectFolderItem(
+				TestPropsValues.getUserId(),
+				_objectDefinition4.getObjectDefinitionId(),
+				_objectFolderB.getObjectFolderId(), 0, 0);
+
+		_objectFolderItemLocalService.deleteObjectFolderItem(objectFolderItem);
+
+		Assert.assertNull(
+			_objectFolderItemLocalService.fetchObjectFolderItem(
+				objectFolderItem.getObjectFolderItemId()));
+	}
+
+	@Test
+	public void testDeleteObjectFolderItemByObjectDefinitionId() {
+		Assert.assertNotNull(
+			_objectFolderItemLocalService.
+				getObjectFolderItemsByObjectDefinitionId(
+					_objectDefinition1.getObjectDefinitionId()));
+
+		_objectFolderItemLocalService.
+			deleteObjectFolderItemByObjectDefinitionId(
+				_objectDefinition1.getObjectDefinitionId());
+
+		Assert.assertEquals(
+			Collections.emptyList(),
+			_objectFolderItemLocalService.
+				getObjectFolderItemsByObjectDefinitionId(
+					_objectDefinition1.getObjectDefinitionId()));
+	}
+
+	@Test
+	public void testDeleteObjectFolderItemByObjectFolderId() {
+		Assert.assertNotNull(
```

Comment:
I'm thinking if we should inline this method 🤔

Reply:
This method reflect a method existing in LocalService, [see](https://github.com/liferay/liferay-portal/blob/8219becef1c2a9ed47827298da92fc970fb25116/modules/apps/object/object-service/src/main/java/com/liferay/object/service/impl/ObjectFolderItemLocalServiceImpl.java#L115-L123).